### PR TITLE
optimize: use TreeMap instead of the LinkedHashMap in TableMeta to compatible high level MySQL

### PR DESCRIPTION
**Pull request by caohdgege**
### Ⅰ. Describe what this PR did
optimize: use TreeMap instead of LinkedHashMap  to compatible high level MySQL

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by sustly — APPROVED** [review]
LGTM

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMeta.java [hunk 1]
```diff
@@ -39,11 +39,11 @@
      * key: column name
      */
 
-    private Map<String, ColumnMeta> allColumns = new LinkedHashMap<>();
+    private Map<String, ColumnMeta> allColumns = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```
Comment (l81893521) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMeta.java:42]: What s the purpose of LinkedHashMap change to TreeMap?

Comment (caohdgege) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMeta.java:42]: Map#get & Map#containsKey of TreeMap will ignore case

**Review by slievrly — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**